server/src/main/java/org/elasticsearch/gateway/GatewayService.java

```diff
@@ -131,8 +129,14 @@ public void clusterChanged(final ClusterChangedEvent event) {
         }
 
         final ClusterState state = event.state();
+        final DiscoveryNodes nodes = state.nodes();
 
-        if (state.nodes().isLocalNodeElectedMaster() == false) {
+        if (nodes.getMasterNodeId() == null) {
```

Comment:
I think this is already covered by the check on `nodes.isLocalNodeElectedMaster()`.

Reply:
Yep I kept it because it has a separate logging message. I have now moved it inside `isLocalNodeElectedMaster` check to retain the logging message.